isotope/convert/pkg/graph/script/command.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this currentFile except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package script
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// Command is the top level interface for commands.
+type Command interface{}
+
+const (
+	sleepCommandKey   = "sleep"
+	requestCommandKey = "call"
+)
+
+func commandsToMarshallable(cmds []Command) ([]interface{}, error) {
+	marshallableCmds := make([]interface{}, 0, len(cmds))
+	for _, cmd := range cmds {
+		marshallableCmd, err := commandToMarshallable(cmd)
+		if err != nil {
+			return marshallableCmds, err
+		}
+		marshallableCmds = append(marshallableCmds, marshallableCmd)
+	}
+	return marshallableCmds, nil
+}
+
+func commandToMarshallable(cmd Command) (marshallable interface{}, err error) {
+	switch cmd := cmd.(type) {
+	case SleepCommand:
+		marshallable = map[string]string{sleepCommandKey: cmd.String()}
+	case RequestCommand:
+		marshallable = map[string]RequestCommand{requestCommandKey: cmd}
+	case ConcurrentCommand:
+		marshallable, err = commandsToMarshallable(cmd)
```

Comment:
is partial return useful or needed?

Reply:
I believe this is addressed by removing named returns.